macros/parserGraphTool.pl

```diff
@@ -524,14 +524,14 @@ sub cmp_preprocess {
 		my $graphObjs = @{$self->{staticObjects}} ?
 			join(",", @{$self->{staticObjects}}, $ans->{student_ans}) : $ans->{student_ans};
 
-		# This first ends the attempts table MathJax_Preview script.  Note that the script
-		# started here is ended by the original script end tag for the MathJax_Preview.
+		# This first ends the attempts table MathJax math mode, then injects the jsxgraph div and javascript, and then
+		# starts another MathJax math mode that is ended by the atttempts table.  Both of the MathJax elements are
+		# removed by the injected javascript.
 		$ans->{preview_latex_string} = <<"END_ANS";
-</script>
+\\]
 <div id='${ans_name}_student_ans_graphbox' class='graphtool-answer-container'></div>
 <script>
-jQuery("#${ans_name}_student_ans_graphbox").parent().find('span').remove();
-jQuery("#${ans_name}_student_ans_graphbox").parent().find('script[type^="math/tex"]').remove();
+jQuery("#${ans_name}_student_ans_graphbox").parent().find('mjx-container').remove();
```

Comment:
I don't think you want to remove this container, as some of the the MathJax CSS is tied to this.  It is also what sets the size, adds the MathJax menu, and so on.  Can you tell me what the purpose is?  I think the original removals were for the preview span and the script that contained the original TeX code.  MathJax 3 doesn't include either of those.

Reply:
The original removal was to remove the preview span and script, as you said, although it wouldn't contain any TeX code.  The result before the javascript acted would have been
```html
<span class="MathJax_Preview">[math]</span><script type="math/tex; mode=display"></script>
<div id='${ans_name}_correct_ans_graphbox' class='graphtool-answer-container'></div>
<script>
jQuery("#${ans_name}_correct_ans_graphbox").parent().find('span').remove();
jQuery("#${ans_name}_correct_ans_graphbox").parent().find('script[type^="math/tex"]').remove();
...
</script>
```
Then the javascript removed the span and type="math/tex" script.  The reason for this was that MathJax created an empty element that added space above the graph.  Furthermore, if you click on the graph it shows the popover twice.

Now the reason for the javascript is much the same.  The result with this pull request before the javascript acts is
```html
\[\]
<div id='${ans_name}_correct_ans_graphbox' class='graphtool-answer-container'></div>
<script>
jQuery("#${ans_name}_correct_ans_graphbox").parent().find('mjx-container').remove();
...
</script>
\[\]
```
With this, MathJax again creates empty elements where both of the `\[\]` occur.  The javacript "attempts" to remove those elements.  The reason is those elements add space above and below the graph, as well as the double popover issue.

The point here is that there is no TeX.  The answers are the javascript graphs.  Yes, my original code and this replacement code are hacks.

Now, when I tested this originally this was working, and the mjx-container elements were removed, but now when I test this those elements are not being removed.  There is probably a race condition occurring between when the MathJax typesetting javascript is run and when this jQuery ready function injected here is run.  So I will need to do something different.  I have a fix that I will commit shortly.  I am going to add a flag to the answer hash that is checked in the attempts table that tells it this is not a TeX preview, so don't add the TeX delimiters.